### ISSUE TITLE
Swipe from left to right to go back 

### ISSUE THREAD

**Issue by GaniduAsh** [biobeats/swiftui-navigation-stack#8]
Trying to implement a custom navigation as the NavigationView that SwiftUI supports doesnt provide many features, 

Notice the one in here is quite impressive. But I noticed this doesnt have the feature to go back to the previous screen by dragging left to right (default go back feature that iOS provides)

Is there a way to get it done ?

**Comment by matteopuc** [comment]
Hi @GaniduAsh , thanks for asking. Unfortunately, at the moment, there's no easy way to get the "swipe to pop" feature with `NavigationStackView`. Anyway, it's a very interesting feature to work on.

**Comment by GaniduAsh** [comment]
Yeah i Noticed it.  I think the reason is having your own navigation stack as its hard to maintain all the features that default iOS navigation has, hence i used UIViewControllerRepresentable and the stack which comes with UINavigation.

**Comment by divyanshunegi** [comment]
I would like to have that swipe to pop gesture, and would help to implement the same, I am new to SwiftUI coming from Android, but not having this swipe gesture does not feel like iOS to me when implementing this Navigation Stack, apart from that it works smooth and easy to integrate.

**Comment by GaniduAsh** [comment]
But its the default iOS behavior and its a must feature in apple . I would
suggest to integrate it as i mentioned earlier

**Comment by JuanVeGu-Ionix** [comment]
Is there any solution for this case?

**Comment by matteopuc** [comment]
@JuanVeGu-Ionix Unfortunately not yet.

**Comment by JuanVeGu-Ionix** [comment]
> @JuanVeGu-Ionix Unfortunately not yet.

To solve this, for the moment, I get the swipe gesture to the right and pass it the pop() method 😬

**Comment by ln-12** [comment]
@JuanVeGu-Ionix but I think this approach is not the same as the default iOS behavior. I mean you cannot swipe the current screen back and forth, right?

Would also love to see this feature!

**Comment by matteopuc** [comment]
@JuanVeGu-Ionix It would be great if you showed us how you've implemented this feature. We could find a way to integrate your implementation in this project. Thanks.

**Comment by ln-12** [comment]
It's probably something like [this](https://stackoverflow.com/a/59109283/13440564) where the `pop()` method is called in the `.right` case. But as I said, this does not replicate the iOS gesture which also includes swiping the current screen only partly.

**Comment by JuanVeGu-Ionix** [comment]
> It's probably something like [this](https://stackoverflow.com/a/59109283/13440564) where the `pop()` method is called in the `.right` case. But as I said, this does not replicate the iOS gesture which also includes swiping the current screen only partly.

Yes, I created a ViewModifier with the next structured

´´´
struct BackToView: ViewModifier {
    let function: () -> Void
    
    func body(content: Content) -> some View {
        content.gesture(DragGesture(minimumDistance: 20, coordinateSpace: .global).onEnded { value in
            let horizontalAmount = value.translation.width as CGFloat
            let verticalAmount = value.translation.height as CGFloat
            
            if abs(horizontalAmount) > abs(verticalAmount) && horizontalAmount > 0 {
                function()
            }
        })
    }
}
´´´
